src/vm/precode.cpp

```diff
@@ -521,7 +545,7 @@ TADDR Precode::AllocateTemporaryEntryPoints(MethodDescChunk *  pChunk,
 #ifdef HAS_COMPACT_ENTRYPOINTS
     // Note that these are just best guesses to save memory. If we guessed wrong,
     // we will allocate a new exact type of precode in GetOrCreatePrecode.
-    BOOL fForcedPrecode = pFirstMD->RequiresStableEntryPoint(count > 1);
+    BOOL fForcedPrecode = hasMethodDescVersionableWithPrecode || pFirstMD->RequiresStableEntryPoint(count > 1);
```

Comment:
The call to `RequiresStableEntryPoint(true /* estimateForChunk */)` would estimate for the chunk, but the estimation would not necessarily correct for the versionability check. This is the only place where `true` is passed for the argument, the code here tries to compensate for that. For untiered methods the estimate doesn't have to be correct but tiered methods require a backpatchable precode.

Reply:
Actually the estimation doesn't have to be correct anyway, a wrong estimation would just cause additional precodes to be allocated instead of sharing one backpatchable precode along with the temporary entry point. Probably doesn't matter much.